dm/syncer/syncer.go

```diff
@@ -1497,69 +1541,53 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
 		flushCheckpoint = true
 		err = s.loadTableStructureFromDump(ctx)
 		if err != nil {
-			tctx.L().Warn("error happened when load table structure from dump files", zap.Error(err))
+			s.tctx.L().Warn("error happened when load table structure from dump files", zap.Error(err))
 			cleanDumpFile = false
 		}
 	} else {
 		cleanDumpFile = false
 	}
 
+	s.runWg.Add(1)
+	// start syncDML worker first because if there is no dml worker checkpointFlushWorker will blocked forever
```

Comment:
why we don't this this order in original code?

Reply:
this  case was founded when i hack code ,i want to see all thread lifetime in s.Run so i just commnet the `go s.syncDM` but if `syncDML` not start, `s.Run` will pending on `checkpointFlushWorker.run` as a result dm-woker can not exit any more

this case can not hanppen in real world,  but i perfer to start `syncDML` thread before call `flushCheckPoints` because there is actually an order of priority between

Reply:
since in real scenario this won't happen, I prefer we delete this comment and line 1555, it only causes confusion